test/conformance/ingress/headers_test.go

```diff
@@ -0,0 +1,304 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"knative.dev/serving/pkg/apis/networking"
+	"knative.dev/serving/pkg/apis/networking/v1alpha1"
+	"knative.dev/serving/test"
+	"knative.dev/serving/test/types"
+	v1a1test "knative.dev/serving/test/v1alpha1"
+)
+
+// TestPreSplitSetHeaders verifies that an Ingress that specified AppendHeaders pre-split has the appropriate header(s) set.
+func TestPreSplitSetHeaders(t *testing.T) {
+	t.Parallel()
+	clients := test.Setup(t)
+
+	name, port, cancel := CreateService(t, clients, networking.ServicePortNameHTTP1)
+	defer cancel()
+	test.CleanupOnInterrupt(cancel)
+
+	const headerName = "Foo-Bar-Baz"
+
+	// Create a simple Ingress over the Service.
+	ing := &v1alpha1.Ingress{
```

Comment:
With the next test I'm working on I am actually factoring a function that takes `(t, clients, spec)` and has a variant that will wait for it to be ready and return an `http.Client` to dramatically cut down boilerplate.

I'll update this once that lands because this is so flaky with Istio that I doubt we should land it 🙄